scanpy/utils.py

```diff
@@ -563,6 +569,183 @@ def unique_categories(categories):
     categories = np.array(natsorted(categories, key=lambda v: v.upper()))
     return categories
 
+# --------------------------------------------------------------------------------
+# Plotting data helpers
+# --------------------------------------------------------------------------------
+
+# TODO: implement diffxpy method, make singledispatch
+def rank_genes_groups_df(
+    adata: AnnData,
+    group: str,  # Can this be something other than a str?
+    *,
+    key: str = "rank_genes_groups",
+    pval_cutoff: Optional[float] = None,
+    logfc_min: Optional[float] = None,
+    logfc_max: Optional[float] = None,
+    gene_symbols: Optional[str] = None
+) -> pd.DataFrame:
+    """
+    Get `rank_genes_groups` results in the form of a :class:`pd.DataFrame`.
+
+    Params
+    ------
+    adata
+        Object to get results from.
+    group
+        Which group (key from :func:`scanpy.tl.rank_genes_groups` `groupby`) to
+        return results from.
+    key
+        Key differential expression groups were stored under.
+    pval_cutoff
+        Minimum adjusted pval to return.
+    logfc_min
```

Comment:
After some discussions, we thought that `log2FC` is the most unambiguous and most commonly used name, and we would rename the slot this way in `v2.0` (https://github.com/theislab/scanpy/issues/453#issue-404768505). Do you agree? Would you adapt the parameter names?

Reply:
Sure! No strong opinion on this, as long as we're definitely calculating a log2 fold change.

Reply:
Yes, we're definitely calculating log2. I think this is an established convention.

Reply:
`log2fc` or `log2FC`? My pinkies would prefer less caps, but I'd probably also tab complete it most times.

Reply:
I'm fine with `log2fc` but I think diffxpy uses `log2FC` and if you think about an axes label, the capped version might be more appealing. Your call.

Reply:
But, please update https://github.com/theislab/scanpy/issues/453#issue-404768505. 🙂

Reply:
It looks like diffxpy uses `"log2fc"`, at least as of https://github.com/theislab/diffxpy/commit/0054f900aa9569ec00da5aa30876d8ee9d1d3092, so I think I'll go with that